### PR TITLE
feat: add support for extrinsic v5

### PR DESCRIPTION
**Pull request by carlosala**
This PR should not be merged YET!

There are open threads on https://github.com/paritytech/polkadot-sdk/pull/3685 and `merkleize-metadata` / `ledger-signer` has to be confirmed.

### CODE REVIEW

**Review by tiennma2k23 — APPROVED** [review]

**Comment by pandres95** [comment]
@carlosala are there any updates regarding this PR? Since last october 18th, https://github.com/paritytech/polkadot-sdk/pull/3685 was merged, and there are ongoing efforts to introduce some contains extensions that would use the `General` format (https://github.com/paritytech/polkadot-sdk/pull/8164, https://github.com/virto-network/frame-contrib/issues/27).

So, it'd be awesome to have a version that allows testing these capabilities, yet at the same time is somewhat stable.

**Comment by carlosala** [comment]
Hey! Outside of the runtime, extrinsic V5 is blocked by Metadata V16, which isn't stable (yet). Nevertheless, you should be able to create your own signers to use the new extrinsic version.

**Comment by pandres95** [comment]
I noticed that so far [polkadot-sdk#8139](https://github.com/paritytech/polkadot-sdk/issues/8139) has been published, marking the stabilization of `V16` by 30th April, and while not easily visible (I had to open every link manually 😅), all of the issues behind [polkadot-sdk#4520](https://github.com/paritytech/polkadot-sdk/issues/4520) for `V16` have been closed so far.

Additionally, I checked, and for every closed issue, its corresponding merged PR is already released on a stable version (or via backport), which means the completion is ready.

- https://github.com/paritytech/polkadot-sdk/issues/4519 closed by [polkadot-sdk#5274](https://github.com/paritytech/polkadot-sdk/pull/5274) (PRDoc [here](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2503/prdoc/stable2412/pr_5274.prdoc)).
- https://github.com/paritytech/polkadot-sdk/issues/4098 closed by [polkadot-sdk#4948](https://github.com/paritytech/polkadot-sdk/pull/4948) (PRDoc [here](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2503/prdoc/stable2412/pr_4851.prdoc)[^1]).
- https://github.com/paritytech/polkadot-sdk/issues/5285 closed by [polkadot-sdk#5732](https://github.com/paritytech/polkadot-sdk/pull/5732) (PRDoc [here](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2503/prdoc/stable2412/pr_5732.prdoc)).
- https://github.com/paritytech/polkadot-sdk/issues/5347 also closed by [polkadot-sdk#5732](https://github.com/paritytech/polkadot-sdk/pull/5732) (PRDoc [here](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2503/prdoc/stable2412/pr_5732.prdoc)).
- https://github.com/paritytech/polkadot-sdk/issues/5468 closed in favour of https://github.com/paritytech/polkadot-sdk/issues/4519 (already mentioned above).
- https://github.com/paritytech/polkadot-sdk/issues/5980 also closed by [polkadot-sdk#5732](https://github.com/paritytech/polkadot-sdk/pull/5732) (PRDoc [here](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2503/prdoc/stable2412/pr_5732.prdoc)).
- https://github.com/paritytech/polkadot-sdk/issues/6833 closed by [polkadot-sdk#7412](https://github.com/paritytech/polkadot-sdk/pull/7412) and [polksdot-sdk#4722](https://github.com/paritytech/polkadot-sdk/pull/4722) (PRDocs [here](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2503/prdoc/stable2503/pr_7412.prdoc) and [here](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2503/prdoc/stable2503/pr_4722.prdoc)).
- https://github.com/paritytech/polkadot-sdk/issues/7352 closed by [polkadot-sdk#7607](https://github.com/paritytech/polkadot-sdk/pull/7607) (PRDoc [here](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2503/prdoc/stable2503/pr_7607.prdoc)).

In conclusion, I guess it'd be safe to move forward with, at least, a pre-release of this feature.

What are your thoughts?

cc/ @jsdw, @lexnv

[^1]: https://github.com/paritytech/polkadot-sdk/pull/4948 is an extension of https://github.com/paritytech/polkadot-sdk/pull/4851, hence they share the same PRDoc file.

**Comment by jsdw** [comment]
@pandres95 this issue merging will signal the "stabilisation" of V16 metadata now:

https://github.com/paritytech/polkadot-sdk/pull/8443#issuecomment-2855145647

**Comment by carlosala** [comment]
Indeed, I started working with Metadata V16 in #1025. Nevertheless, if metadata V16 is available, signers can be adapted to extrinsic v5 with little work.
We won't do it by ourselves in the main `polkadot-api`, though, until [Extrinsic V5 RFC](https://github.com/polkadot-fellows/RFCs/pull/124) is reviewed adequately by several peers, and merged.

**Comment by pandres95** [comment]
@carlosala is there a practical guide about how to adapt the signer to do this? We have a transaction extension that does the signing part, so we'll start using `General` transactions, but we need to create that signer. 😅 

Furthermore, is there an Element, or Discord channel where I can request for some assistance at some point? I'm starting to exhaustively use PAPI in my developments, and I suspect at any point I'll exceed what's available on the docs (like the question above) probably too often.

**Review by voliva — APPROVED** [review]
